sdk/batch/batch-rest/samples-dev/quick-start.ts

```diff
@@ -0,0 +1,156 @@
+    contentType: "application/json; odata=minimalmetadata",
+  };
+  const result = await client.path("/pools").post(poolParams);
+  if (isUnexpected(result)) {
+    console.log(result.body);
+    throw new Error(`Failed to create pool: ${result.body.message}`);
+  }
+  console.log(`Pool ${poolName} created`);
+}
+
+async function createJob(client: BatchClient) {
+  const jobAddParam: CreateJobParameters = {
+    body: {
+      id: jobId,
+      poolInfo: { poolId: poolName },
+    },
+    contentType: "application/json; odata=minimalmetadata",
+  };
+
+  const result = await client.path("/jobs").post(jobAddParam);
+  if (isUnexpected(result)) {
+    throw new Error(`Failed to create job: ${result.body.message}`);
+  }
+  console.log(`Job ${jobId} created`);
+}
+
+async function submitTasks(client: BatchClient) {
+  for (let i = 1; i < 6; i++) {
+    const addTaskParam: CreateTaskParameters = {
+      body: {
+        id: `task${i}`,
+        commandLine: `cmd /c echo task-${i} > result.txt`,
+      },
+      contentType: "application/json; odata=minimalmetadata",
+    };
+
+    const addTaskResult = await client.path("/jobs/{jobId}/tasks", jobId).post(addTaskParam);
+    if (isUnexpected(addTaskResult)) {
+      throw new Error(`Failed to add task ${i}: ${addTaskResult.body.message}`);
+    }
+    console.log(`Task ${i} added`);
+  }
+}
+
+async function listTasks(client: BatchClient) {
+  const result = await client
+    .path("/jobs/{jobId}/tasks", jobId)
+    .get({ queryParameters: { maxresults: 1 } });
+  if (isUnexpected(result)) {
+    throw new Error(`Failed to list tasks: ${result.body.message}`);
+  }
+  console.log("Tasks first result:", result.body.value?.length);
+  const taskAsyncIterator = paginate(client, result);
+
+  for await (const task of taskAsyncIterator) {
+    console.log(`- ${task.id}, state: ${task.state}`);
+  }
+}
+
+async function waitForTasks(client: BatchClient) {
+  console.log("Waiting for tasks to complete");
+  while (true) {
+    const result = await client
+      .path("/jobs/{jobId}/tasks", jobId)
+      .get({ queryParameters: { maxresults: 1 } });
+    if (isUnexpected(result)) {
+      throw new Error(`Failed to list tasks: ${result.body.message}`);
+    }
+    const taskAsyncIterator = paginate(client, result);
+    let isAllCompleted = true;
+    for await (const task of taskAsyncIterator) {
+      if (task.state !== "completed") {
+        console.log(`- ${task.id}, state: ${task.state}`);
+        isAllCompleted = false;
+        break;
+      }
+    }
+    if (isAllCompleted) {
+      console.log("All tasks completed");
+      break;
+    }
+    await new Promise((resolve) => setTimeout(resolve, 5000));
+  }
+}
+
+async function cleanup(client: BatchClient) {
+  const result = await client.path("/pools/{poolId}", poolName).delete();
+  if (isUnexpected(result)) {
+    throw new Error(`Failed to delete pool: ${result.body.message}`);
+  }
+  console.log(`Pool ${poolName} deleted`);
+}
+
+main();
```

Comment:
better to add something to indicate completion/error case
```suggestion
main().catch(console.error);
```